lib/asciidoctor-mallard/converter.rb

```diff
@@ -217,17 +217,21 @@ def stem node
 
     def olist node
       result = []
+      element_name = (node.role? 'steps') ? 'steps' : 'list'
       type_attribute = %( type="#{OLIST_TYPES[node.style]}")
+      if element_name == 'steps' && type_attribute == ' type="numbered"'
```

Comment:
My preference here would be to not compare against type_attribute directly, as that seems prone to breakage in the future. Maybe instead we could do something more like:

```ruby
olist_type = OLIST_TYPES[node.style]
type_attribute = element_name == 'steps' && olist_type == 'numbered' ? nil : %( type="#{olist_type}")
```
?

Reply:
I applied this change after merging.